src/rules/eva-voice-biometric.js

```diff
@@ -0,0 +1,143 @@
+/**
+*	@title EVA Voice Biometric connector
+*	@overview EVA Voice Biometric connector rule for Auth0 enables voice enrolment and verification as a second factor
+*	@gallery true
+*	@category marketplace
+*	@author Auraya Systems 2020
+*
+*	EVA WEB is a second-factor voice biometric solution designed for use with a range of web browsers that support audio capture. 
+*	The biometric verification process is intended to be part of an authentication flow to enhance security, privacy and ease-of-use. 
+*	EVA WEB is an offering from Auraya Systems.
+*
+*	The voice tokens used with EVA WEB are predominantly digit strings, although phrases are also supported. 
+*	The usual verification process will usually prompt the user to read out a sequence of random digits, 
+*	a sequence of consecutive digits, and optionally their phone number.
+*
+*	The first time a user is directed to EVA WEB where they will be invited to enrol their voice. 
+*	This process captures between 3 and 8 utterances depending on the required voice tokens, and includes a biometric “Opt-In” consent step. 
+*	At the conclusion of enrolment they will be redirected back to the authentication flow.
+*	
+*	Configuration Items:
+*	EVA_URL = EVA endpoint, typically: https://eva-web.mydomain.com/server/oauth
+*	CLIENT_ID = JWT client id on the EVA server (and this server)
+*	CLIENT_SECRET = JWT client secret on the EVA server (and this server)
+*	ISSUER = this app (or "issuer")
+*	
+*	RANDOM_DIGITS = true|false
+*	COMMON_DIGITS = true|false
+*	PERSONAL_DIGITS = a user.user_metadata field that contains digits such as phone_number, or blank
+*/
+function(user, context, callback) {
+
+	console.log(user);
+	//console.log(context);
+	console.log(configuration);
+
+	var clientSecret = configuration.CLIENT_SECRET;
+    
+  
+	// Prepare Users' enrolment status
+	user.user_metadata = user.user_metadata || {};
+	user.user_metadata.eva = user.user_metadata.eva || {};
+
+
+	// User has initiated a login and is prompted to use voice biometrics
+	// Send user's information and query params in a JWT to avoid tampering
+	function createToken(clientId, clientSecret, issuer, user) {
+		const options = {
+			expiresInMinutes: 2,
+			audience: clientId,
+			issuer: issuer
+		};
+
+		return jwt.sign(user, clientSecret, options);
+	}
+
+	if (context.protocol === "redirect-callback") {
+		// User was redirected to the /continue endpoint with correct state parameter value
+
+		var options = {
+			//subject: user.user_id, // validating the subject is nice to have but not strictly necessary
+			jwtid: user.jti
```

Comment:
I want the jti to exist until "redirect-callback", not persisted beyond that - since it is just there to validate the temporary token. If this property is lost when the rule reenters, then I'm not sure why it is still working. Would 'context' be a better object to store these kinds of temporary properties?